### ISSUE TITLE
Extend with custom iterables

### ISSUE THREAD

**Issue by jmakeig** [jmakeig/iterable#8]
`Iterable` is more like an interface. It should be possible for concrete instances to extend this with properties/methods that are specific to their own type, override existing implementations, or fallback to the built-in functionality. 

For example, for a MarkLogic `Sequence`, the `slice()` implementation uses `fn.subsequence()` internally. How can we decouple this from the `Iterable` interface? The `xpath()` method is an extension to the `Iterable` interface and likely not germain to other `Iterable` instances.

`Symbol.species` might help here.

**Comment by jmakeig** [comment]
It won't be possible to cover every iterable in a single implementation. Rather `Iterable` should supply a default naïve implementation and then Take a look at an [example inheritance technique](https://gist.github.com/jmakeig/45cf2fa19103f8a870f11895f5350624).

``` js
function Iterable(iterable) { … }
Iterable.prototype = Object.assign({},
   { slice(start, end): { … } }
);
function SequenceIterable() { … }
SequenceIterable.prototype = Object.assign(
  Object.create(Iterable.prototype),
  {
    slice(start, end): {
      …
    }
  }
);

```

**Comment by jmakeig** [comment]
Need to update docs.

**Comment by jmakeig** [comment]
Closed in 1436ac234e7a807826d3cc810664b4b20a3bba56.
